rpm/0005-Add-checks-to-fs_private_dir_mount.patch

```diff
@@ -0,0 +1,40 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Tomi=20Lepp=C3=A4nen?= <tomi.leppanen@jolla.com>
+Date: Mon, 22 Feb 2021 09:59:55 +0200
+Subject: [PATCH] Add checks to fs_private_dir_mount
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+Check that the directory exists before attempting to mount it.
+
+Signed-off-by: Tomi Leppänen <tomi.leppanen@jolla.com>
+---
+ src/firejail/fs_etc.c | 11 +++++++++++
+ 1 file changed, 11 insertions(+)
+
+diff --git a/src/firejail/fs_etc.c b/src/firejail/fs_etc.c
+index 908134ef..7234c6e2 100644
+--- a/src/firejail/fs_etc.c
++++ b/src/firejail/fs_etc.c
+@@ -188,6 +188,17 @@ void fs_private_dir_copy(const char *private_dir, const char *private_run_dir, c
+ }
+ 
+ void fs_private_dir_mount(const char *private_dir, const char *private_run_dir) {
++	assert(private_dir);
++	assert(private_run_dir);
++
++	// nothing to do if directory does not exist
++	struct stat s;
++	if (stat(private_dir, &s) == -1) {
++		if (arg_debug)
++			printf("Cannot find %s\n", private_dir);
```

Comment:
Moving this to happen after "intent to mount" debug logging few lines below would give this diagnostic message a better context. And emitting also failure reason (e.g. "Cannot find %s: %m\n") might be helpful for debugging in case it is permissions / stte instead of directory simply not existing.

Reply:
Probably better to use strerror() instead of %m for portability though. It seems they want to run this on libc libraries too. But in any case, good suggestion, thanks!